### PR TITLE
update UT registrycontroller class

### PR DESCRIPTION
**Pull request by vivekNEC**
we can not merge in the development yet because other controller class UT is remeaning.

### CODE REVIEW

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java [hunk 1]
```diff
@@ -38,13 +44,16 @@
...
+import eu.neclab.ngsildbroker.commons.datatypes.DBWriteTemplates;
 import eu.neclab.ngsildbroker.commons.datatypes.requests.AppendEntityRequest;
 import eu.neclab.ngsildbroker.commons.datatypes.requests.CreateEntityRequest;
 import eu.neclab.ngsildbroker.commons.datatypes.requests.EntityRequest;
 import eu.neclab.ngsildbroker.commons.datatypes.requests.UpdateEntityRequest;
 import eu.neclab.ngsildbroker.commons.datatypes.results.UpdateResult;
 import eu.neclab.ngsildbroker.commons.enums.ErrorType;
 import eu.neclab.ngsildbroker.commons.exceptions.ResponseException;
+import eu.neclab.ngsildbroker.commons.storage.StorageDAO;
+import eu.neclab.ngsildbroker.commons.tools.HttpUtils;
```
Comment (amitrnec) [inline Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java]: [Trival]remove unused packages.

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java [hunk 2]
```diff
@@ -229,6 +257,7 @@ public void updateMessageTest() throws Exception {
 			Map<String, Object> entityBody = (Map<String, Object>) JsonUtils.fromString(entityPayload);
 			UpdateEntityRequest request = new UpdateEntityRequest(multimaparr, "urn:ngsi-ld:Vehicle:A103", entityBody,
 					resolved, null);
+
```
Comment (amitrnec) [inline Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java:309]: [Trival] fix indentation.

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java [hunk 3]
```diff
@@ -395,10 +338,11 @@ public void updatePartialDefaultAttributeFieldTest() {
 		try {
 			ArrayListMultimap<String, String> entityIds = ArrayListMultimap.create();
 			entityIds.put(AppConstants.INTERNAL_NULL_KEY, "urn:ngsi-ld:Vehicle:A103");
-			when(entityInfoDAO.getAllIds()).thenReturn(entityIds);
-			Method postConstruct = EntityService.class.getDeclaredMethod("loadStoredEntitiesDetails");
-			postConstruct.setAccessible(true);
-			postConstruct.invoke(entityService);
+			// when(entityInfoDAO.getAllIds()).thenReturn(entityIds);
```
Comment (amitrnec) [inline Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java]: [Trival] remove commented code..

File: Registry/RegistryManager/pom.xml [hunk 4]
```diff
@@ -105,5 +105,17 @@
 			<groupId>org.postgresql</groupId>
 			<artifactId>postgresql</artifactId>
 		</dependency>
+		<dependency>
```
Comment (amitrnec) [inline Registry/RegistryManager/pom.xml:108]: [Major] please remove dependency from here and place in common pom.

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java [hunk 5]
```diff
@@ -1,148 +1,476 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+//import java.util.Optional;
```
Comment (amitrnec) [inline Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java]: [Trival] remove commented code.

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java [hunk 6]
```diff
@@ -1,148 +1,476 @@
...
+@RunWith(SpringRunner.class)
+//@WebMvcTest
+@AutoConfigureMockMvc//(secure = false) 
+public class RegistryControllerTest {
+
+	@Autowired
+	private MockMvc mockMvc;
+	@MockBean
+	private CSourceService csourceService;
+
```
Comment (amitrnec) [inline Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java]: [Trival] remove space

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java [hunk 7]
```diff
@@ -1,181 +1,290 @@
...
+			List<String> entities = new ArrayList<String>();
+			result.put(AppConstants.INTERNAL_NULL_KEY, entities);
+			when(csourceInfoDAO.getAllEntities()).thenReturn(result);
+			ReflectionTestUtils.setField(csourceService, "AUTO_REG_STATUS", "active");
+			Method postConstruct = CSourceService.class.getDeclaredMethod("loadStoredEntitiesDetails"); // methodName,parameters
+			postConstruct.setAccessible(true);
+			postConstruct.invoke(csourceService);
+			multimaparr.put("content-type", "application/json");
+			when(csourceInfoDAO.getEntity(any(), any())).thenReturn(payload);
+			boolean deleteresult = csourceService.deleteEntry(multimaparr,
```
Comment (amitrnec) [inline Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:270]: [Trival] remove unused code same as other.

File: Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java [hunk 8]
```diff
@@ -1,146 +1,280 @@
...
+		subscription.setId("urn:ngsi-ld:Subscription:173223");
+		subscription.setType("Subscription");
+		subscription.setNotification(notifyParam);
+		multimaparr.put("content-type", "application/json");
+		SubscriptionRequest subRequest = new SubscriptionRequest(subscription, context, multimaparr, 0);
+		when(subscriptionId2Subscription.get(any())).thenReturn(new Subscription());
+		try {
+			manager.updateSubscription(subRequest);
+		} catch (Exception e) {
+			verify(tenant2subscriptionId2Subscription, times(1)).get(any(), any());
```
Comment (amitrnec) [inline Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java]: [Minor] why you handle verify inside the catch block.

File: Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java [hunk 9]
```diff
@@ -1,146 +1,280 @@
...
+		subscription.setNotification(notifyParam);
+		multimaparr.put("content-type", "application/json");
+		ArrayList<String> result = new ArrayList<String>();
+
+		SubscriptionRequest subRequest = new SubscriptionRequest(subscription, context, multimaparr, 0);
+		tenant2subscriptionId2Subscription.put(subRequest.getTenant(), subRequest.getId().toString(), subRequest);
+		ReflectionTestUtils.setField(baseSubscriptionService, "tenant2subscriptionId2Subscription",
+				tenant2subscriptionId2Subscription);
+
+		// PowerMockito.spy(BaseSubscriptionInfoDAO.class);
```
Comment (amitrnec) [inline Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java]: [Trival] remove unused code

File: Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java [hunk 10]
```diff
@@ -1,146 +1,280 @@
...
+		tenant2subscriptionId2Subscription.put(subRequest.getTenant(), subRequest.getId().toString(), subRequest);
+		ReflectionTestUtils.setField(baseSubscriptionService, "tenant2subscriptionId2Subscription",
+				tenant2subscriptionId2Subscription);
+
+		try {
+			manager.getSubscription("urn:ngsi-ld:Subscription:173223", multimaparr);
+		} catch (ResponseException e) {
+			errorMessage = e.getMessage();
+		}
+		Assert.assertEquals(errorMessage, "urn:ngsi-ld:Subscription:173223 not found");
```
Comment (amitrnec) [inline Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java]: [Major] why you compare with "not found" but I think it should be return id.

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java [hunk 11]
```diff
@@ -1,181 +1,401 @@
...
+			Gson gson = new Gson();
+			Map<String, Object> resolved = gson.fromJson(updatePayload, Map.class);
+			Map<String, Object> originalresolved = gson.fromJson(payload, Map.class);
+			String[] optionsArray = new String[0];
+			AppendCSourceRequest request = new AppendCSourceRequest(multimaparr,
+					"urn:ngsi-ld:ContextSourceRegistration:csr3", originalresolved, resolved, optionsArray);
+			when(csourceInfoDAO.getEntity(any(), any())).thenReturn(payload);
+			updateResult = csourceService.appendToEntry(multimaparr, "urn:ngsi-ld:ContextSourceRegistration:csr3",
+					resolved, optionsArray);
+			Assert.assertEquals(updateResult.getUpdated(), request.getUpdateResult().getUpdated());
```
Comment (amitrnec) [inline Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:243]: [Major] as per the method name indication, This test case for check the id exist on not but you have compared object but it should be compare like 404 or resource not found.

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java [hunk 12]
```diff
@@ -1,181 +1,401 @@
...
+			postConstruct.setAccessible(true);
+			postConstruct.invoke(csourceService);
+			multimaparr.put("content-type", "application/json");
+			String registrationid = "urn:ngsi-ld:ContextSourceRegistration:csr3";
+			when(csourceInfoDAO.getEntity(any(), any()))
+					.thenThrow(new ResponseException(ErrorType.NotFound, registrationid + " not found"));
+			try {
+				csourceService.getCSourceRegistrationById(null, "urn:ngsi-ld:ContextSourceRegistration:csr4");
+			} catch (Exception e) {
+				Assert.assertEquals(registrationid + " not found", e.getMessage());
```
Comment (amitrnec) [inline Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:363]: can you justify why use compare in catch block..

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java [hunk 13]
```diff
@@ -1,181 +1,401 @@
...
+		} catch (Exception ex) {
+			Assert.fail();
+		}
+	}
+
+	/**
+	 * this method is use for get CSource Registry
+	 */
+	@Test
+	public void getCSourceRegistrationTest() throws Exception {
```
Comment (amitrnec) [inline Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:375]: [Trival] please change method name as per the test case.

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java [hunk 14]
```diff
@@ -1,433 +1,605 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
```
Comment (amitrnec) [inline Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java]: [Trival] remove unused packages/space.

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java [hunk 15]
```diff
@@ -1,433 +1,605 @@
...
+  
+	}
+
+	@After
+	public void tearDown() {
+		entity = null;
+		response = null;
+	}
+
+	@Test
```
Comment (amitrnec) [inline Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java:154]: [Minor] Add comment in each method..

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java [hunk 16]
```diff
@@ -1,433 +1,605 @@
...
+    
+	@Test
+	public void getEntityByIdBadRequestTest() throws Exception {
+		try {
+
+			QueryResult result = new QueryResult(entitiesBadRequest, null, ErrorType.None, -1, true);
+			  result.setActualDataString(entitiesBadRequest);
+				when(queryService.getData(any(), any(), any(), any(), any()))
+				.thenThrow(new ResponseException(ErrorType.BadRequestData, "invalid query")).thenReturn(result);
+
```
Comment (amitrnec) [inline Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java:213]: [Trival] remove unused space from everywhere

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java [hunk 17]
```diff
@@ -1,433 +1,605 @@
...
+	  result.setActualDataString(entities);
+	  when(queryService.getData(any(), any(), any(), any(), any())).thenReturn(result);
+	  ResultActions resultAction=	mockMvc.perform(get("/ngsi-ld/v1/types")
+				.accept(AppConstants.NGB_APPLICATION_JSON)).andExpect(status().isOk())
+				.andExpect(jsonPath("$.id").value("urn:ngsi-ld:Vehicle:A101"));
+		MvcResult mvcResult = resultAction.andReturn();
+		MockHttpServletResponse response = mvcResult.getResponse();
+		int status = response.getStatus();
+		verify(queryService, times(1)).getData(any(), any(), any(), any(), any());
+	assertEquals(200, status);
```
Comment (amitrnec) [inline Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java]: [Trival] Fix indentation.

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java (hunk 1 above)
Comment (pawannec) [reply Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java]: done

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java (hunk 2 above)
Comment (pawannec) [reply Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java:309]: Done

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java (hunk 3 above)
Comment (pawannec) [reply Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java]: Done

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java (hunk 7 above)
Comment (pawannec) [reply Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:270]: Done

File: Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java (hunk 9 above)
Comment (pawannec) [reply Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java]: Done

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java [hunk 18]
```diff
@@ -208,117 +240,102 @@ public void createMessageThrowsAlreadyExistTest() throws ResponseException, Exce
...
 			Gson gson = new Gson();
-			Map<String, Object> resolved = gson.fromJson(updatePayload, Map.class);
+			Map<String, Object> resolved = gson.fromJson(updatePayloadforwrongattr, Map.class);
 			Map<String, Object> entityBody = (Map<String, Object>) JsonUtils.fromString(entityPayload);
 			UpdateEntityRequest request = new UpdateEntityRequest(multimaparr, "urn:ngsi-ld:Vehicle:A103", entityBody,
 					resolved, null);
 			updateResult = entityService.updateEntry(multimaparr, "urn:ngsi-ld:Vehicle:A103", resolved);
-			Assert.assertEquals(request.getUpdateResult().getUpdated(), updateResult.getUpdated());
-			verify(entityService).updateEntry(any(), any(), any());
+			Assert.assertEquals(request.getUpdateResult().getNotUpdated(), updateResult.getNotUpdated());
```
Comment (amitrnec) [inline Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java:260]: [Minor] please check this assert equals condition  I think we need to compare with message..

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java (hunk 5 above)
Comment (vivekNEC) [reply Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java]: done

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java (hunk 6 above)
Comment (vivekNEC) [reply Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/controller/RegistryControllerTest.java]: Done

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java (hunk 14 above)
Comment (vivekNEC) [reply Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java]: Done

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java (hunk 15 above)
Comment (vivekNEC) [reply Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java:154]: Done

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java (hunk 16 above)
Comment (vivekNEC) [reply Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java:213]: Done

File: Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java (hunk 17 above)
Comment (vivekNEC) [reply Core/QueryManager/src/test/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryControllerTest.java]: Done

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java (hunk 12 above)
Comment (pawannec) [reply Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:363]: it throws response exception if id is not match

File: Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java (hunk 10 above)
Comment (pawannec) [reply Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java]: it throws response exception so we compare inside catch block

File: Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java (hunk 8 above)
Comment (pawannec) [reply Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java]: Remove verify and compare actual and expected message

**Review by amitrnec — APPROVED** [review]

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java (hunk 13 above)
Comment (pawannec) [reply Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:375]: done

File: Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java (hunk 11 above)
Comment (pawannec) [reply Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java:243]: in this case update result return array list object so for this we compare array list result

File: Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java (hunk 18 above)
Comment (pawannec) [reply Core/EntityManager/src/test/java/eu/neclab/ngsildbroker/entityhandler/services/EntityServiceTest.java:260]: It throws array list object so we not compare with message

**Review by ScorpioBroker — APPROVED** [review]

**Review by amitrnec — APPROVED** [review]

**Pull request merged**